### ISSUE TITLE
Re-confirm email address if a user changes it

### ISSUE THREAD

**Issue by imphil** [librecores/librecores-web#211]
If a user changes his email address the new address must be validated again.

Possible approach: https://blog.intelligentbee.com/2015/08/24/how-to-enable-email-confirmation-on-fosuserbundle-profile-edit/

**Comment by sandipbhuyan** [comment]
@imphil I think the application doesn't have the facility to change the email address. So one issue should be created to add that feature.

**Comment by eliaskousk** [comment]
@sandipbhuyan We do have this facility with fos user bundle.
Currently the route is `domain/profile/edit` and it will be changed to `user/profile/edit` once #193 is merged.

This issue should be tied to that PR which I will rename to 'E-Mail & Password Reset'.

**Comment by sandipbhuyan** [comment]
@eliaskousk I am getting that page when I am entering the route in the url. My question is, is there any way to go to that link from the UI?

**Comment by eliaskousk** [comment]
@sandipbhuyan I see your point and you are right we don't have a link to the profile edit on the current UI. This probably has to do with the fact that we use a [user profile settings](https://www.librecores.org/user/settings/profile) page for such stuff and not the `domain/profile/` routes from fos user bundle.

Thus instead of activating / renaming the `domain/profile` route/s we should  actually keep it disabled and add a `Change E-Mail` tab above or below the [Change Password](https://www.librecores.org/user/settings/password) tab of `domain/user/settings/` route .

**Comment by sandipbhuyan** [comment]
@eliaskousk would I create a Issue for that?

**Comment by eliaskousk** [comment]
@sandipbhuyan thanks, but let me first work on #193 during the weekend because I may address this there instead of opening a new issue. @imphil what do you think?

**Comment by imphil** [comment]
@eliaskousk sounds good, I think both approaches would work.

Back to this bug: There seems to be some movement on the FOSUserBundle side to get the re-validation of emails feature into upstream, but these efforts seems to have failed for now.  PR with the code changes (reverted in https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2731): https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2612#issuecomment-366206464

So if we cannot figure out an easy way to re-send the confirmation mails we could just backlog this for now.

**Comment by azine** [comment]
For anyone still interested in the "update email confirmation" feature: as FOSUserBundle does not move, I we implemented this feature as separate bundle.
See https://github.com/azine/AzineEmailUpdateConfirmationBundle
